src/chai-form.ts

```diff
@@ -289,8 +289,40 @@ export class ChaiForm extends LitElement {
 
   override connectedCallback() {
     super.connectedCallback();
+    this.verifyCurrentFlowInstanceIdThroughFormLoad();
+  }
 
-    api(this.environment).formLoad(localStorage.getItem('chai-visitorId')!, this.flowType, localStorage.getItem('chai-flowInstanceId'));
+  private verifyCurrentFlowInstanceIdThroughFormLoad() {
+    // Avoid running any field updates while this is running
+    console.debug('Form connected, running FormLoad', this.formInstanceId);
+    localStorage.setItem('chai-form-load', String(Date.now()));
+    api(this.environment)
+      .formLoad(
+        localStorage.getItem('chai-visitorId')!,
+        this.flowType,
+        localStorage.getItem('chai-flowInstanceId')
+      )
+      .then((localStorageValid) => {
+        console.debug('FormLoad finished', this.formInstanceId);
+        localStorage.removeItem('chai-form-load');
+        if (!localStorageValid) {
+          console.info('Resetting form state', this.formInstanceId);
+          if (localStorage.getItem('chai-flowInstanceId') != null) {
+            console.warn(
+              'Removing flow instance ID from local storage',
+              this.formInstanceId,
+              this.formInstanceId
+            );
+          }
+          const fieldElements = this.getFieldsInCurrentSlot();
+          fieldElements.forEach((element) => {
+            // This way we reset the fields in localStorage and in the form
+            (element as ChaiFieldBase<unknown>).reset();
+          });
+          localStorage.removeItem('chai-flowInstanceId');
+          localStorage.removeItem('chai-load-time-flow-instance');
```

Comment:
Extract lines 309-323 (the "what you do if `!localStorageValid`) to a private `reset` method.